sql-odbc/src/odfesqlodbc/es_info.cpp

```diff
@@ -476,8 +492,28 @@ void SetTableTuples(QResultClass *res, const TableResultSet res_type,
         // information for its Data Preview window.
         std::string catalog("");
         bind_tbl[TABLES_CATALOG_NAME]->UpdateData((void *)catalog.c_str(), 0);
-        for (size_t i = 0; i < binds.size(); i++)
-            binds[i]->AssignData(&tuple[i]);
+
+        for (size_t i = 0; i < binds.size(); i++) {
+            // Add tuples for SQLColumns
+            if (binds.size() > COLUMNS_SQL_DATA_TYPE) {
```

Comment:
Should we just report the extra columns regardless?

Reply:
The same method is used while adding tuples for SQLTables as well. SQLTables has only 5 columns whereas SQLColumns has 18. 
So this check ensures that the data type column is updated for SQLColumns only. If we remove this check then we get Memory Corruption since SQLTables will try to access column which doesn't exist.

Reply:
Almost wondering if it would be worth it to split some of this logic up, now that we're handling special cases for SQLTables & SQLColumns

Reply:
Created issue #630 to revisit the logic of assigning data for SQLTables & SQLColumns